src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -121,7 +121,7 @@ class DirectedEdge
                                      << target_shift)}
       , weight{weight}
   {}
-  KOKKOS_FUNCTION constexpr DirectedEdge() = default;
+  KOKKOS_DEFAULTED_FUNCTION constexpr DirectedEdge() = default;
```

Comment:
FYI https://github.com/kokkos/kokkos/pull/5371#discussion_r951950151
I don't care much but so you know eventually the right thing to do is have nothing.